internal/expand/tree.go

```diff
@@ -130,6 +134,10 @@ func (t *Tree) ToProto() *acl.SubjectTree {
 
 // swagger:ignore
 func TreeFromProto(t *acl.SubjectTree) (*Tree, error) {
+	if t == nil {
+		return nil, nil
```

Comment:
Is passing a nil value here not an error?

Reply:
That depends on whether the expand engine should never return nil. Currently there are two cases:

1. [depth of zero](https://github.com/ory/keto/blob/40563ccb84e966c2feca06c5bc5e247d987afd18/internal/expand/engine.go#L29)
2. [no such relation tuple exists](https://github.com/ory/keto/blob/40563ccb84e966c2feca06c5bc5e247d987afd18/internal/expand/engine.go#L56)

But that can be fixed, although I don't think they are an error.

Reply:
Ok, as long as the code is able to deal with nil trees downstream I think this is fine. Is that tested?

Reply:
Not too much (that's why I found it right now), I will add some more tests.